internal error when trying to add a companion

using Paper 1.18.1 

whenever I try to add someone as a companion it says "An internal error occurred while attempting to perform this command"

```
11.01 03:56:47 [Server] INFO DumbAnimations issued server command: /tardis add conallator
11.01 03:56:47 [Server] ERROR null
11.01 03:56:47 [Server] INFO org.bukkit.command.CommandException: Unhandled exception executing command 'tardis' in plugin TARDIS v4.8.1-b2388
11.01 03:56:47 [Server] INFO at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
11.01 03:56:47 [Server] INFO at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
11.01 03:56:47 [Server] INFO at org.bukkit.craftbukkit.v1_18_R1.CraftServer.dispatchCommand(CraftServer.java:897) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.server.network.ServerGamePacketListenerImpl.handleCommand(ServerGamePacketListenerImpl.java:2285) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.server.network.ServerGamePacketListenerImpl.handleChat(ServerGamePacketListenerImpl.java:2096) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.server.network.ServerGamePacketListenerImpl.handleChat(ServerGamePacketListenerImpl.java:2077) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.network.protocol.game.ServerboundChatPacket.handle(ServerboundChatPacket.java:46) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.network.protocol.game.ServerboundChatPacket.a(ServerboundChatPacket.java:6) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:56) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1413) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.c(MinecraftServer.java:189) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1391) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1384) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:110) ~[?:?]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1518) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1264) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.18.1.jar:git-Paper-145]
11.01 03:56:47 [Server] INFO at java.lang.Thread.run(Thread.java:833) ~[?:?]
11.01 03:56:47 [Server] INFO Caused by: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "org.bukkit.OfflinePlayer.getName()" is null
11.01 03:56:47 [Server] INFO at me.eccentric_nz.TARDIS.utility.TARDISStaticUtils.getOfflinePlayer(TARDISStaticUtils.java:294) ~[TARDIS.jar:?]
11.01 03:56:47 [Server] INFO at me.eccentric_nz.TARDIS.commands.tardis.TARDISAddCompanionCommand.doAdd(TARDISAddCompanionCommand.java:97) ~[TARDIS.jar:?]
11.01 03:56:47 [Server] INFO at me.eccentric_nz.TARDIS.commands.tardis.TARDISCommands.onCommand(TARDISCommands.java:102) ~[TARDIS.jar:?]
11.01 03:56:47 [Server] INFO at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[paper-api-1.18.1-R0.1-SNAPSHOT.jar:?]
11.01 03:56:47 [Server] INFO ... 21 more
```

this also happened before when I tried to use just regular Spigot

--- a/src/main/java/me/eccentric_nz/TARDIS/utility/TARDISStaticUtils.java
+++ b/src/main/java/me/eccentric_nz/TARDIS/utility/TARDISStaticUtils.java
@@ -305,7 +305,7 @@ public class TARDISStaticUtils {
      */
     public static OfflinePlayer getOfflinePlayer(UUID uuid) {
         for (OfflinePlayer player : Bukkit.getOfflinePlayers()) {
-            if (player.getUniqueId().equals(uuid)) {
+            if (uuid.equals(player.getUniqueId())) {
                 return player;
             }
         }

--- a/src/main/java/me/eccentric_nz/TARDIS/utility/TARDISStaticUtils.java
+++ b/src/main/java/me/eccentric_nz/TARDIS/utility/TARDISStaticUtils.java
@@ -214,8 +214,8 @@ public class TARDISStaticUtils {
     }
 
     public static String getNick(UUID uuid) {
-        if (Bukkit.getServer().getPluginManager().getPlugin("Essentials") != null) {
-            Essentials essentials = (Essentials) Bukkit.getServer().getPluginManager().getPlugin("Essentials");
+        Essentials essentials = (Essentials) Bukkit.getServer().getPluginManager().getPlugin("Essentials");
+        if (essentials != null) {
             User user = essentials.getUser(uuid);
             String prefix = essentials.getSettings().getNicknamePrefix();
             return ChatColor.stripColor(user.getNick(false)).replace(prefix, "");
@@ -230,8 +230,8 @@ public class TARDISStaticUtils {
     }
 
     public static String getNick(Player player) {
-        if (Bukkit.getServer().getPluginManager().getPlugin("Essentials") != null) {
-            Essentials essentials = (Essentials) Bukkit.getServer().getPluginManager().getPlugin("Essentials");
+        Essentials essentials = (Essentials) Bukkit.getServer().getPluginManager().getPlugin("Essentials");
+        if (essentials != null) {
             User user = essentials.getUser(player.getUniqueId());
             String prefix = essentials.getSettings().getNicknamePrefix();
             return ChatColor.stripColor(user.getNick(false)).replace(prefix, "");
@@ -291,7 +291,7 @@ public class TARDISStaticUtils {
      */
     public static OfflinePlayer getOfflinePlayer(String name) {
         for (OfflinePlayer player : Bukkit.getOfflinePlayers()) {
-            if (player.getName().equals(name)) {
+            if (player.getName() != null && player.getName().equals(name)) {
                 return player;
             }
         }

--- a/src/main/java/me/eccentric_nz/TARDIS/utility/TARDISStaticUtils.java
+++ b/src/main/java/me/eccentric_nz/TARDIS/utility/TARDISStaticUtils.java
@@ -291,7 +291,7 @@ public class TARDISStaticUtils {
      */
     public static OfflinePlayer getOfflinePlayer(String name) {
         for (OfflinePlayer player : Bukkit.getOfflinePlayers()) {
-            if (player.getName() != null && player.getName().equals(name)) {
+            if (name.equals(player.getName())) {
                 return player;
             }
         }
@@ -311,5 +311,4 @@ public class TARDISStaticUtils {
         }
         return null;
     }
-
 }

